lib/matplotlib/colorbar.py

```diff
@@ -510,166 +490,288 @@ def __init__(self, ax, cmap=None,
+        linewidths : float or array-like
+            Either a single linewidth applying to all lines or one linewidth
+            for each line.
+        erase : bool, default: True
+            Whether to remove any previously added lines.
+        """
+        y = self._locate(levels)
+        rtol = (self._y[-1] - self._y[0]) * 1e-10
+        igood = (y < self._y[-1] + rtol) & (y > self._y[0] - rtol)
+        y = y[igood]
+        if np.iterable(colors):
+            colors = np.asarray(colors)[igood]
+        if np.iterable(linewidths):
+            linewidths = np.asarray(linewidths)[igood]
+        X, Y = np.meshgrid([self._y[0], self._y[-1]], y)
         if self.orientation == 'vertical':
-            long_axis, short_axis = ax.yaxis, ax.xaxis
-            if mpl.rcParams['ytick.minor.visible']:
-                self.minorticks_on()
+            xy = np.stack([X, Y], axis=-1)
         else:
-            long_axis, short_axis = ax.xaxis, ax.yaxis
-            if mpl.rcParams['xtick.minor.visible']:
-                self.minorticks_on()
-        long_axis.set(label_position=self.ticklocation,
-                      ticks_position=self.ticklocation)
-        short_axis.set_ticks([])
-        short_axis.set_ticks([], minor=True)
+            xy = np.stack([Y, X], axis=-1)
+        col = collections.LineCollection(xy, linewidths=linewidths,
+                                         colors=colors)
+
+        if erase and self.lines:
+            for lc in self.lines:
+                lc.remove()
+            self.lines = []
+        self.lines.append(col)
+
+        # make a clip path that is just a linewidth bigger than the axes...
+        fac = np.max(linewidths) / 72
+        xy = np.array([[0, 0], [1, 0], [1, 1], [0, 1], [0, 0]])
+        inches = self.ax.get_figure().dpi_scale_trans
+        # do in inches:
+        xy = inches.inverted().transform(self.ax.transAxes.transform(xy))
+        xy[[0, 1, 4], 1] -= fac
+        xy[[2, 3], 1] += fac
+        # back to axes units...
+        xy = self.ax.transAxes.inverted().transform(inches.transform(xy))
+        if self.orientation == 'horizontal':
+            xy = xy.T
+        col.set_clip_path(mpath.Path(xy, closed=True),
+                          self.ax.transAxes)
+        self.ax.add_collection(col)
         self.stale = True
 
-    config_axis = _api.deprecate_privatize_attribute("3.3")
+    def update_ticks(self):
+        """
+        Setup the ticks and ticklabels. This should not be needed by users.
+        """
+        ax = self.ax
+        # Get the locator and formatter; defaults to self.locator if not None.
+        self._get_ticker_locator_formatter()
+        self._long_axis().set_major_locator(self.locator)
+        self._long_axis().set_minor_locator(self.minorlocator)
+        self._long_axis().set_major_formatter(self.formatter)
 
     def _get_ticker_locator_formatter(self):
         """
         Return the ``locator`` and ``formatter`` of the colorbar.
```

Comment:
🤷‍♂️   Maybe as a follow on.  I'm not sure why this was made public in the first place, and I figure this will be invasive enough that I didn't want to try and deprecate things I didn't fully understand.